.github/workflows/main.yml

```diff
@@ -290,6 +290,15 @@ jobs:
         # ensure that the tests fail more quickly so that people can fix them.
         timeout-minutes: 30
 
+      # Since tests run in the context of the container, we need to copy all
+      # the coverage reports out of the container and into the host filesystem.
+      - name: Copy coverage reports to the host
+        run: ./tools/copy_docker_coverage_reports.sh test_container
```

Comment:
This is the only moderately complex thing going on in this PR. Because tests are run in a Docker container, we need to copy the files back to the host and put them in the right places for Codecov's Action to be able to do the right thing.